### PR TITLE
Fix for #46

### PR DESCRIPTION
**Pull request by farzadshbfn**
I re-implemented LinkedList in a non-recursive format.

I also re-implemented DoublyLinkedList implementation with non-recursive format, but this relies on the fact that does compiler optimize calling of `array` on `LinkedList` because of it's `tail-recursion` or not.

### CODE REVIEW

File: Sources/LinkedList.swift [hunk 1]
```diff
@@ -9,17 +9,32 @@ class LinkedList<T> {
     }
 
     init?(array: [T]) {
+        let reversed = Array(array.reversed())
```
Comment (wokalski) [inline Sources/LinkedList.swift]: Maybe it's possible to use the result of `array.reversed()` directly?

**Review by wokalski — CHANGES_REQUESTED** [review]
Everything's awesome

File: Sources/LinkedList.swift (hunk 1 above)
Comment (farzadshbfn) [reply Sources/LinkedList.swift]: Fixed it (thanks to your `Collection` extension)

**Review by wokalski — APPROVED** [review]

**Comment by wokalski** [comment]
Travis is just super slow today 👎. I will merge (or you can) when CI is done.

Thank you for your contributions. This project would've been much worse without your help ✊

**Comment by farzadshbfn** [comment]
🙏🏼🙏🏼

**Pull request merged**